docs/pages/Cardinal/API-Reference/Transactions.mdx

```diff
@@ -0,0 +1,208 @@
+# Transactions
+
+This API reference will help you understand how to use transactions and configuration options available for transactions.
+
+```go
+package transactions
+
+import "pkg.world.dev/world-engine/cardinal/ecs"
+
+// AttackInput is the input for the Attack transaction.
+type AttackInput struct {
+    TargetID uint64
+}
+
+// AttackOutput is the output for the Attack transaction.
+type AttackOutput struct {
```

Comment:
Can we call this `AttackReceipt`

Reply:
`Receipts` are already defined as:

```go
// Receipt contains a transaction ID, an arbitrary result, and a list of errors.
type Receipt struct {
	ID     transaction.TxID
	Result any
	Errs   []error
}
```

I think naming this `AttackReceipt` might be confusing. Additionally, the signature for `NewTransactionType`'s type parameters is `In, Out`, so i used `Output` to stay consistent there.

Reply:
to be clear, output is going to do the `Result` field of the receipt struct right?

if that's the case, i think naming it AttackReceipt makes sense because it goes into the receipt struct

Reply:
i think the fact that it goes specifically into a field named `Result` lets itself better to being named `AttackResult` - since `Receipt` for us is more broad (errors & result)

Reply:
I really want to use AttackReceipt here because I want to be very clear that this is going into the receipt. It's not intuitive from `AttackResult` where that result is going.

Reply:
I'll go ahead and change it (strong opinion weakly held) - we may want to consider more renamings here though - as we now have situations like this:

```go
AttackTx.SetResult(w, txID, AttackReceipt{Success: true})
```

perhaps we want to change this to method `SetReceipt`? or maybe its fine as is. just want to draw attention to the mismatch there.

Reply:
yeah, i think we should create a ticket to that. Should be SetReceipt

Reply:
I agree with Tyler, in that we should have SetResult on the transaction, and that "result" should go into the "result" field in a struct called "receipt". To me, a "result" is what you get after processing a transaction in some happy path. I haven't seen "receipt" used much in software development, so coming up with our own definition (a result and associated errors) feels right.

Reply:
That also feels intuitively right here to me too^, @smsunarto wdyt?

Reply:
can we revisit here? cc @smsunarto

Reply:
Can we call it `SetReceiptResult `

Reply:
Meaning we would also have AddReceiptError? 

I prefer `SetResult` and `AddError`

Reply:
ok, i'll agree with you guys here